clients/src/test/java/org/apache/kafka/clients/consumer/internals/TopicMetadataRequestManagerTest.java

```diff
@@ -151,6 +152,36 @@ void testHardFailures(Exception exception) {
         }
     }
 
+    @Test
+    public void testNetworkTimeout() {
```

Comment:
I expected such a test also in the other request manager tests. The only request manager test that has such test other than this is `HeartbeatRequestManager`.

Reply:
Per previous comment: NetworkTimeout isn't handled correctly in the commitRequestManager, therefore I want to address this in that PR.  I will check other request manager to ensure this is tested.

Reply:
CoordinatorRequestManager doesn't have it - so I added the test.

Reply:
Not entirely sure about the behavior of the OffsetRequestManager - I'll need to get back to this.  Added a ticket there: KAFKA-15642